firmware/hw_layer/debounce.h

```diff
@@ -5,17 +5,36 @@
  * @date Aug 31, 2020
  * @author David Holdeman, (c) 2020
  */
+#ifndef DEBOUNCE_INC
+#define DEBOUNCE_INC
+
 #include "globalaccess.h"
 #include "io_pins.h"
 
+class ButtonDebounce;
+
+class PointerListNode {
+public:
+    ButtonDebounce *pointer = NULL;
+    PointerListNode *next = NULL;
+};
+
 class ButtonDebounce {
 public:
-    void init(int t, brain_pin_e p, pin_input_mode_e m);
+    void init(int t, DECLARE_CONFIG_POINTERS(brain_pin_e, p), DECLARE_CONFIG_POINTERS(pin_input_mode_e, m));
+    void updateConfiguration();
     bool readPinEvent();
+    PointerListNode thisNode;
 private:
     int threshold;
     efitick_t timeLast;
-    brain_pin_e pin;
-    iomode_t mode;
+    brain_pin_e *pin;
+    brain_pin_e *active_pin;
+    brain_pin_e oldPin;
+    pin_input_mode_e *mode;
+    pin_input_mode_e *active_mode;
     bool readValue;
```

Comment:
"= false" just to be be nice? it's nice to have default state for as many fields as reasonable?